src/java.net.http/share/classes/jdk/internal/net/http/HttpConnection.java

```diff
@@ -76,17 +78,48 @@ abstract class HttpConnection implements Closeable {
     public static final Comparator<HttpConnection> COMPARE_BY_ID
             = Comparator.comparing(HttpConnection::id);
 
+    private static final AtomicLong LABEL_COUNTER = new AtomicLong();
+
     /** The address this connection is connected to. Could be a server or a proxy. */
     final InetSocketAddress address;
     private final HttpClientImpl client;
     private final TrailingOperations trailingOperations;
+
+    /**
+     * A unique identifier that provides a total order among instances.
+     */
     private final long id;
 
-    HttpConnection(InetSocketAddress address, HttpClientImpl client) {
+    /**
+     * A label to identify the connection.
+     * <p>
+     * This label helps with associating multiple components participating in a
+     * connection. For instance, an {@link AsyncSSLConnection} and the
+     * {@link PlainHttpConnection} it wraps will share the same label. As a
+     * result, compared to {@link #id}, this label does not give a total order
+     * among instances.
```

Comment:
```suggestion
    /**
     * A label to identify the connection.
     * <p>
     * This label helps with associating multiple components participating in a
     * connection. For instance, an {@link AsyncSSLConnection} and the
     * {@link PlainHttpConnection} it wraps will share the same label.
```

Reply:
Implemented in 42ff0fc4a83cd33f1a35e4a2b3e7ce5c25ea39cc.